core/src/swarm/swarm.rs

```diff
@@ -258,7 +265,11 @@ where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
                     self.behaviour.inject_node_event(conn_info.peer_id().clone(), event);
                 },
                 Async::Ready(RawSwarmEvent::Connected { conn_info, endpoint }) => {
-                    self.behaviour.inject_connected(conn_info.peer_id().clone(), endpoint);
+                    if self.banned_peers.contains(conn_info.peer_id()) {
+                        self.raw_swarm.peer(conn_info.peer_id().clone()).into_connected().unwrap().close();
```

Comment:
We try to do like Substrate/Polkadot and use `expect` with a reason.
```suggestion
                        self.raw_swarm.peer(conn_info.peer_id().clone())
                            .into_connected()
                            .expect("the RawSwarm just notified us that we were connected; QED")
                            .close();
```